channeldb/mp_payment.go

```diff
@@ -177,6 +178,18 @@ func (m *MPPayment) InFlightHTLCs() []HTLCAttempt {
 	return inflights
 }
 
+// GetAttempt returns the specified htlc attempt on the payment.
+func (m *MPPayment) GetAttempt(id uint64) (*HTLCAttempt, error) {
+	for _, htlc := range m.HTLCs {
+		htlc := htlc
+		if htlc.AttemptID == id {
```

Comment:
This is why I wanted the map...

Reply:
could always add an internal index on `MPPayment`

Reply:
Yes that is true. I have to admit that I do like the ordering on the payment htlcs. But could also have been a simple sort in `rpcserver`.